### ISSUE TITLE
Allow trailing punctuation on same line after math display

### ISSUE THREAD

**Issue by onomou** [typst/typst#3936]
### Description

In many math textbooks, a centered equation may be followed by punctuation. This allows the completion of a sentence that ends with the equation. Here are some examples:

NCSSM Contemporary Precalculus 3e
![image](https://github.com/typst/typst/assets/131693/03ea5e11-50f0-4ce9-a641-719f9a07f1c8)

OpenStax Calculus vol. 2
![image](https://github.com/typst/typst/assets/131693/28a12b78-1477-45ca-b4ad-7b1c7be40aae)

The standard way to replicate this in both LaTeX and Typst is to put the punctuation inside math mode (see [Period after math-mode equation](https://tex.stackexchange.com/questions/32853/period-after-math-mode-equation)). This comes with its own problems, as Werner explains in their response:

> Naturally, this may lead to a slightly off-center spacing due to the punctuation (albeit marginally visible). However, to correct for this, you can use \phantom{<punct>} on the opposite side:

I would love to see Typst do better.

```typst
For the private jet to be twice as far from Dallas as the commercial jet is,
$
2(1 - 1/3 t) &= 1 - 1/3.5 t\
2 - 2/3 t &= 1 - 1/3.5 t\
1 &= 2/3 t - 1/3.5 t\
10.5 &= 7 t - 3 t\
10.5 &= 4 t\
2.625 &= t\
2 inline(5/8) &= t\
$.
```

### Use Case

This would be useful for maintaining separation between normal text and math text. Putting the punctuation inside math mode makes the equation code impure, which in turn makes it necessary to modify the equation when copying it into another context.

**Comment by Enivex** [comment]
I don't agree with the off-center comment. The punctuation is part of the equation, so if anything it would look off-center if disregarded.

That being said, if that would be desirable, I do not see why that means the punctuation has to go outside the equation.

**Comment by MDLC01** [comment]
I think this is a duplicate of #1474

**Comment by Enivex** [comment]
> I think this is a duplicate of #1474

Indeed it is

**Comment by onomou** [comment]
Well that's embarrassing. Somehow I missed #1474. My thought regarding purity is that an equation may be useful outside its original context, and it would be less frustrating to be able to directly copy-paste or otherwise insert it elsewhere without modifying it. Here is an example:

```typst
#let myeq = {
  $ x^2+2x+1=0 $
}

In the context of our discussion, #myeq.

Elsewhere, #myeq does not need the punctuation.
```

**Comment by Enivex** [comment]
> Well that's embarrassing. Somehow I missed #1474. My thought regarding purity is that an equation may be useful outside its original context, and it would be less frustrating to be able to directly copy-paste or otherwise insert it elsewhere without modifying it. Here is an example:
> 
> ```typst
> #let myeq = {
>   $ x^2+2x+1=0 $
> }
> 
> In the context of our discussion, #myeq.
> 
> Elsewhere, #myeq does not need the punctuation.
> ```

Vastly more often than not, you should be referencing an equation instead of reusing it. It would also only be possible for the special case of an equation with no internal punctuation.
